diff --git a/node_modules/mongoose/types/query.d.ts b/node_modules/mongoose/types/query.d.ts
index 1d66e75..f1560e9 100644
--- a/node_modules/mongoose/types/query.d.ts
+++ b/node_modules/mongoose/types/query.d.ts
@@ -694,6 +694,11 @@ declare module 'mongoose' {
      */
     set(path: string | Record<string, unknown>, value?: any): this;
 
+    /**
+     * Adds a `$unset` to this query's update without changing the operation
+     */
+    unset(path: string | string[]): this;
+
     /** Sets query options. Some options only make sense for certain operations. */
     setOptions(options: QueryOptions<DocType>, overwrite?: boolean): this;
 
